subgraphs/_reference_/src/sdk/protocols/generic/protocol.ts

```diff
@@ -0,0 +1,250 @@
+import { Versions } from "../../../../../../deployment/context/interface";
+
+import { Protocol as ProtocolSchema } from "../../../../generated/schema";
+import { TokenPricer } from "../config";
+import { Config } from "./config";
+import { ProtocolSnapshot } from "./protocolSnapshot";
+import { AccountWasActive } from "./account";
+import * as constants from "../../util/constants";
+import {
+  dataSource,
+  Address,
+  Bytes,
+  BigDecimal,
+} from "@graphprotocol/graph-ts";
+import { SDK } from ".";
+import { BIGINT_ZERO } from "../lending/constants";
+import { CustomEventType } from "../../util/events";
+
+/**
+ * Bridge is a wrapper around the BridgeProtocolSchema entity that takes care of
+ * safely and conveniently updating the entity. Updating the Bridge entity using this
+ * wrapper also takes care of the Financials and Usage snapshots.
+ */
+export class ProtocolManager {
+  protocol: ProtocolSchema;
+  event: CustomEventType;
+  pricer: TokenPricer;
+  snapshoter: ProtocolSnapshot;
+  sdk: SDK | null = null;
+  /**
+   * Creates a new Bridge instance. This should only be called by the Bridge.load
+   * @private
+   */
+  private constructor(
+    protocol: ProtocolSchema,
+    pricer: TokenPricer,
+    event: CustomEventType
+  ) {
+    this.protocol = protocol;
+    this.event = event;
+    this.pricer = pricer;
+    this.snapshoter = new ProtocolSnapshot(protocol, event);
+  }
+
+  /**
+   * This is the main function to instantiate a Bridge entity. Most times it is not called directly, but from the SDK initializer.
+   *
+   * @param conf {BridgeConfigurer} An object that implements the BridgeConfigurer interface, to set some of the protocol's properties
+   * @param pricer {TokenPricer} An object that implements the TokenPricer interface, to allow the wrapper to access pricing data
+   * @param event {CustomEventType} The event being handled at a time.
+   * @returns Bridge
+   */
+  static load(
+    conf: Config,
+    pricer: TokenPricer,
+    event: CustomEventType
+  ): ProtocolManager {
+    const id = Address.fromString(conf.getID());
+    let protocol = ProtocolSchema.load(id);
+    if (protocol) {
+      const proto = new ProtocolManager(protocol, pricer, event);
+      proto.setVersions(conf.getVersions());
+      return proto;
+    }
+
+    protocol = new ProtocolSchema(id);
+    protocol.name = conf.getName();
+    protocol.slug = conf.getSlug();
+    protocol.network = dataSource.network().toUpperCase().replace("-", "_");
```

Comment:
Should we move this into its own function? And then add something to also replace GNOSIS with XDAI?

Reply:
Actually, if you use `gnosis` to specify network in the subgraph.yaml, then `dataSource.network()` will return `gnosis`. Do you think we need a function for it since this is the case?

Reply:
Hmm, but then the BE will break if it expects `xdai` and you use `gnosis`. No?

I’m thinking, if at a subgraph level we mess up accidentally we can ensure it still works.

Reply:
Even if we don’t replace the gnosis part, I’d still put it in its own function. It does no harm and I’ve seen a few places where we use `datasource.network()` to match with our network enum. I think having it in a central place will work as a reference to refer to when attempting to get the current network. Plus, removing duplicates, which I’ve seen and done myself wrong a few times

Reply:
Yeah that makes sense. I agree then with putting the transformation in a common function. I think, in a way though, it makes sense not to do the conversion. When I was messing with this, the graph-cli caught that I was setting the network value to the protocol that was not in the enum. If we leave it as is, the usage of `xdai` be caught and fixed.

Reply:
Ah right, that makes sense too. Unless someone fixes it by adding it back to the schema :D 
But we might catch that too on the review if anything